### ISSUE TITLE
build_imagenet_data.py requires validation images to be in synset directories

### ISSUE THREAD

**Issue by alexjung** [tensorflow/models#8079]
I want to prepare the ImageNet dataset for training InceptionV3 as described in `research/slim/README.md`. As I have already downloaded the ImageNet dataset for some previous project I wanted to reuse it by not downloading it again but rather just build the required data format for tensorflow. (On a side note, the download with the provided script does not work, as the URLs seem to be out of date.)

Now the problem is that the script `research/slim/datasets/build_imagenet_data.py` requires not only the training images but also the validation images to be in the synset directories like that:
```
data_dir/n01440764/ILSVRC2012_val_00000293.JPEG
data_dir/n01440764/ILSVRC2012_val_00000543.JPEG
```
But they don't come sorted into these directories when you log into image-net.org and download the validation dataset. The training images are sorted into sub-directories, and for those the script worked (I commented the processing of the validation images). The ground truth labels for the validation images come via the devkit download in a .txt file. 

Did anyone else run into this problem? Any thoughts on how to address this apart from writing a script that sorts the validation images into folders based on the ground truth synset labels?

Thanks in advance!

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by alexjung** [comment]
See updated issue description containing the requested information: https://github.com/tensorflow/models/issues/8079#issue-553393806
